soci/soci_index.go

```diff
@@ -256,13 +256,10 @@ func WithPlatform(platform ocispec.Platform) BuildOption {
 	}
 }
 
-// WithLegacyRegistrySupport sets whether the SOCI Index should be built for legacy registries.
-// OCI 1.1 added support for associating artifacts such as SOCI indices with images. There is a
-// mechanism to emulate this behavior with OCI 1.0 registries by pretending that the SOCI index
-// is itself an image. This option should only be use if the SOCI index will be pushed to a
-// registry which does not support OCI 1.1 features.
-func WithLegacyRegistrySupport(c *buildConfig) error {
-	c.legacyRegistry = true
+// WithOCIArtifactRegistrySupport sets whether the SOCI index should be built for registries that
+// support OCI 1.1 artifacts by building an artifact manifest instead of an image manifest.
```

Comment:
nit:

// WithOCIArtifactRegistrySupport sets whether to build SOCI index as an artifact manifest instead of an image
// manifest. Artifact manifest is released in OCI 1.1 spec and is only supported by OCI 1.1 compatible registries.

Reply:
I don't think artifact manifest are officially decided on for 1.1